cpp/src/strings/case.cu

```diff
@@ -59,11 +60,38 @@ struct upper_lower_fn
     character_flags_table_type case_flag; // flag to check with on each character
     const character_flags_table_type* d_flags;
     const character_cases_table_type* d_case_table;
+    const special_case_mapping *d_special_case_mapping;
     const int32_t* d_offsets{};
     char* d_chars{};
 
-    __device__ int32_t operator()(size_type idx)
+    __device__ special_case_mapping get_special_case_mapping_for_code_point( uint32_t code_point )
```

Comment:
If you want to shorten this I think `get_special_case_mapping(uint32_t code_point)` is plenty clear.